content/docs/rfcs/17/README.md

```diff
@@ -153,28 +162,37 @@ message WakuMessage {
-The proof asserts that:
-1. The message publisher is the current member of the group i.e., her/his identity commitment key is part of the membership group Merkle tree with the root `merkle_root`.
-2. `share_x` and `share_y`  are correctly computed.
-3. The `nullifier` is constructed correctly.
 
-Other fields of the `RateLimitProof` message are the public inputs to the rln circuit and used for the generation of the `proof`.
+Other fields of the `RateLimitProof` message are the public inputs to the [rln circuit](/spec/32) and used for the generation of the `proof`.
 
-The `merkle_root` is an array of 32 bytes which holds the root of membership group Merkle tree at the time of publishing the message.
+The `merkle_root` is an array of 32 bytes in little-endian order which holds the root of membership group Merkle tree at the time of publishing the message.
 
-The `epoch` is an array of 32 bytes that represents the epoch in which the message is published.
-<!-- TODO epoch is going to change to a different type -->
+The `epoch` is an array of 32 bytes in little-endian order that represents the epoch in which the message is published.
 
-`share_x` and `share_y` are shares of the user's identity key.
-These shares are created using [Shamir secret sharing scheme](##Publishing). 
+`share_x` and `share_y` are shares of the user's secret identity key `sk`.
 `share_x` is an array of 32 bytes and contains the hash of the `WakuMessage`'s `payload` concatenated with its `contentTopic`. 
-<!-- TODO hash other fields if necessary-->
-`share_y` is also an array of 32 bytes which is calculated using [Shamir secret sharing scheme](##Publishing).
+<!-- We may include other fields in the hash if necessary-->
+`share_y` is also an array of 32 bytes which is calculated using [Shamir secret sharing scheme](/spec/32).
+
+The `nullifier` is the internal nullifier encoded as an array of 32 bytes.
 
-The `nullifier` is an internal nullifier which allows specifying whether two messages are published by the same publisher during the same `epoch`.
-It is an array of 32 bytes.
 
-<!-- TODO to reflect this change on WakuMessage spec once the PR gets mature -->
+# Recommended System Parameters
+## Epoch 
+A sensible value for the epoch depends on the application for which the spam protection is going to be used.
+For example, while the epoch value of `1` second i.e., messaging rate of 1 per second, might be acceptable for a chat application, might be too low for communication among Ethereum network validators.
+One should look at the desired throughput of the application to decide on a proper epoch value.
+In the proof of concept implementation of `17/WAKU-RLN-RELAY` protocol which is available in [nim-waku](https://github.com/status-im/nim-waku), the messaging rate is set to 1 per second i.e. `T = 1 second`.
+Nevertheless, this value is also subject to change depending on user experience.
+
+## Maximum Epoch Gap
```

Comment:
Maybe we can add short paragraph with example of possibly sensible value?

Reply:
Sure, I'll do a bit of research on this matter, especially the sensible clock drift, and will update the RFC with sensible value in another PR.